frontend/src/views/WheresMyGene/components/GeneSearchBar/index.tsx

```diff
@@ -2,43 +2,50 @@ import { Intent } from "@blueprintjs/core";
 import { LoadingIndicator } from "czifui";
 import React, { useCallback, useContext, useMemo } from "react";
 import { EVENTS } from "src/common/analytics/events";
-import { usePrimaryFilterDimensions } from "src/common/queries/wheresMyGene";
+import { usePrimaryFilterDimensions, useFilterDimensions } from "src/common/queries/wheresMyGene";
 import Toast from "src/views/Collection/components/Toast";
 import { DispatchContext, StateContext } from "../../common/store";
 import { selectGenes, selectTissues } from "../../common/store/actions";
 import { Gene } from "../../common/types";
 import Organism from "./components/Organism";
 import QuickSelect from "./components/QuickSelect";
 import { ActionWrapper, Container, LoadingIndicatorWrapper } from "./style";
-
 interface Tissue {
   name: string;
 }
 
 export default function GeneSearchBar(): JSX.Element {
   const dispatch = useContext(DispatchContext);
-  const { selectedGenes, selectedTissues, selectedOrganismId } =
+  const { selectedGenes, selectedTissues, selectedOrganismId, filterDimensions: prevFilterDimensions } =
     useContext(StateContext);
 
   const { data, isLoading } = usePrimaryFilterDimensions();
+  const { data: filterDimensions } = useFilterDimensions();
+  const { tissue_terms } = filterDimensions;
+  const { tissue_terms: prev_tissue_terms } = prevFilterDimensions;
+  const { genes: rawGenes } = data || {};
+
+  const { tissues: rawTissues } = data || {};
+  let tissues = useMemo(() => {
+    if (!rawTissues) return [];
 
-  const { genes: rawGenes, tissues: rawTissues } = data || {};
+    const temp = rawTissues[selectedOrganismId || ""] || [];
+    let tissue_term_names: string[];
+    if (tissue_terms.length === 0) {
```

Comment:
if tissue_terms are an empty array, then we are in the middle of an update. default to the previous filter tissue terms.